torch/nn/functional.py

```diff
@@ -3851,13 +3851,9 @@ def multi_head_attention_forward(query,                           # type: Tensor
     q = q * scaling
 
     if attn_mask is not None:
-
-        # convert ByteTensor attn_mask to float
-        if attn_mask.dtype == torch.uint8:
-            attn_mask = torch.zeros(attn_mask.size(),
-                                    device=attn_mask.device).masked_fill_(attn_mask.to(torch.bool),
-                                                                          float('-inf'))
-
+        assert attn_mask.dtype == torch.float32 or attn_mask.dtype == torch.float64 or \
```

Comment:
Should bool masks be supported as well? bool is trivially convertible to uint8 without copy

Reply:
This if accepts only float32, float64 and uint8. However, later on before passing to `masked_fill_` uint8 mask gets converted to bool. I propose accepting torch.bool masks as well (this would be expected by users, since all other masks in PyTorch are torch.bool only)

Reply:
Fixed.